### PR TITLE
Add a Gazelle plugin to ensure jsonnet files are linted

### PR DESCRIPTION
**Pull request by jrockway**
This adds a new Gazelle plugin that creates `jsonnet_lint_test` rules for any jsonnet files that appear in the repository.  If the file is deleted, the rule is deleted, of course.

I want to do this for shell scripts next, so we can add shellcheck rules for each of them and remove the global shellcheck linter.

The new plugin got some that I missed in the previous PR, as well.  I wish I had written this first.

### CODE REVIEW

File: BUILD.bazel [hunk 1]
```diff
@@ -75,12 +75,14 @@ gazelle_binary(
...
         "@gazelle//language/proto",  # Built-in rule from gazelle for Protos.
         "@rules_python_gazelle_plugin//python",  # Use gazelle from rules_python.
+        "//src/internal/gazelle/jsonnet",
     ],
     visibility = ["//visibility:public"],
 )
 
 # gazelle:prefix github.com/pachyderm/pachyderm/v2
 # gazelle:proto disable_global
+# gazelle:jsonnet_verbose false
```
Comment (bbonenfant) [inline BUILD.bazel:85]: You configure verbosity with a comment?

File: src/internal/gazelle/jsonnet/jsonnet.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+	// Do a final pass over existing rules, to see if there are any that need to be deleted
+	// because the underlying source file was deleted.
+	if args.File != nil {
+		for _, r := range args.File.Rules {
+			if r.Kind() != lintRule {
+				continue
+			}
+			src := r.AttrString("src")
+			if _, ok := seen[src]; !ok {
+				r.DelAttr("src")
```
Comment (bbonenfant) [inline src/internal/gazelle/jsonnet/jsonnet.go:176]: Why is this line necessary? I'm guessing that the following line is actually what deletes the rule?

**Review by bbonenfant — APPROVED** [review]
Very cool!

File: BUILD.bazel (hunk 1 above)
Comment (jrockway) [reply BUILD.bazel:85]: I stole this idea from the nodejs plugin: https://github.com/benchsci/rules_nodejs_gazelle/blob/main/README.md?plain=1#L212

File: src/internal/gazelle/jsonnet/jsonnet.go (hunk 2 above)
Comment (jrockway) [reply src/internal/gazelle/jsonnet/jsonnet.go:176]: Up in Kinds() we declare that a jsonnet_lint_test rule is empty if src is empty.  Since src isn't empty in the existing BUILD.bazel file here, we have to delete it and then tell Gazelle about the rule, so it will go and delete it.
